packages/syft/src/syft/service/action/action_store.py

```diff
@@ -83,23 +84,38 @@ def get(
         return Err(f"Permission: {read_permission} denied")
 
     def get_pointer(
-        self, uid: UID, credentials: SyftVerifyKey, node_uid: UID
+        self,
+        uid: UID,
+        credentials: SyftVerifyKey,
+        node_uid: UID,
     ) -> Result[SyftObject, str]:
         uid = uid.id  # We only need the UID from LineageID or UID
 
         try:
             # 🟡 TODO 34: do we want pointer read permissions?
             if uid in self.data:
                 obj = self.data[uid]
+                read_permission = ActionObjectREAD(uid=uid, credentials=credentials)
                 if isinstance(obj, TwinObject):
-                    obj = (
-                        obj.mock if not is_action_data_empty(obj.mock) else obj.private
-                    )
+                    if self.has_permission(read_permission):
+                        obj = (
+                            obj.mock
+                            if not is_action_data_empty(obj.mock)
```

Comment:
The thing is:
`domain_client.datasets[0].assets[0].data` calls the get() method
`domain_client.datasets[0].assets[0].mock` calls the get_pointer() method
If we the set this to private it will return the private data not the mock to the data owner which is what was breaking the notebook. api/0.8/00-load-data.ipynb